File: frontend/react-paylateapp/src/models/OrderItem.ts
class OrderItem {
    id: number;
    productId: number;
    productName: string;
    quantity: number;
    imgUrl: string;
    price: number;

    constructor(id: number, productId: number, productName: string, quantity: number,
         imgUrl: string, price: number) {
        this.id = id;
        this.productId = productId;
        this.productName = productName;
        this.quantity = quantity;
        this.imgUrl = imgUrl;
        this.price = price;
    }
}

export default OrderItem;

File: backend/order-service/src/main/java/org/payLate/entity/OrderItem.java
package org.payLate.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.Data;

@Entity
@Data
public class OrderItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @ManyToOne
    @JoinColumn(name = "user_order_id")
    @JsonBackReference
    private UserOrder userOrder;

    @Column(name = "product_id")
    private Long productId;

    @Column(name="partner")
    private String partner;

    @Column(name = "quantity")
    private int quantity;

    @Column(name = "img_url")
    private String imgUrl;

    @Column(name = "price")
    private double price;
}